src/Microsoft.AspNet.Mvc.ViewFeatures/ModelStateDictionaryExtensions.cs

```diff
@@ -0,0 +1,76 @@
+    {
+        /// <summary>
+        /// Adds the specified <paramref name="errorMessage"/> to the <see cref="ModelState.Errors"/> instance
+        /// that is associated with the specified <paramref name="expression"/> text key repressentation.
+        /// </summary>
+        /// <typeparam name="TModel">The type of the model.</typeparam>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/> instance this method extends.</param>
+        /// <param name="expression">The expression of <typeparamref name="TModel"/> from
+        /// which text representation will be used as a key of the <see cref="ModelState"/> to add errors to.</param>
+        /// <param name="errorMessage"></param>
+        public static void AddModelError<TModel>(this ModelStateDictionary modelState, Expression<Func<TModel, object>> expression, string errorMessage)
+        {
+            modelState.AddModelError(GetExpressionText(expression), errorMessage);
+        }
+
+        /// <summary>
+        /// Adds the specified <paramref name="exception"/> to the <see cref="ModelState.Errors"/> instance
+        /// that is associated with the specified <paramref name="expression"/> text key repressentation.
+        /// </summary>
+        /// <typeparam name="TModel">The type of the model.</typeparam>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/> instance this method extends.</param>
+        /// <param name="expression">The expression of <typeparamref name="TModel"/> from
+        /// which text representation will be used as a key of the <see cref="ModelState"/> to add errors to.</param>
+        /// <param name="exception">The error message to add.</param>
+        public static void AddModelError<TModel>(this ModelStateDictionary modelState, Expression<Func<TModel, object>> expression, Exception exception)
+        {
+            modelState.AddModelError(GetExpressionText(expression), exception);
+        }
+
+        /// <summary>
+        /// Removes the specified <paramref name="expression"/> text key repressentation from the <see cref="ModelStateDictionary"/>.
+        /// </summary>
+        /// <typeparam name="TModel">The type of the model.</typeparam>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/> instance this method extends.</param>
+        /// <param name="expression">The expression of <typeparamref name="TModel"/> from
+        /// which text representation will be used as a key of the <see cref="ModelState"/> to add errors to.</param>
+        /// <returns>
+        /// true if the element is successfully removed; otherwise, false.
+        /// This method also returns false if expression's text key was not found in the model-state dictionary.
+        /// </returns>
+        public static bool Remove<TModel>(this ModelStateDictionary modelState, Expression<Func<TModel, object>> expression)
+        {
+            return modelState.Remove(GetExpressionText(expression));
```

Comment:
> will remove only the entry for the top-level model and not for any bound property.

Could you elaborate a bit more on this scenario? Because I couldn't completely understand what you meant here.